security/docker/Dockerfile.node-agent-k8s

```diff
@@ -1,5 +1,14 @@
-FROM ubuntu:16.04
-ADD node_agent_k8s /
+# BASE_DISTRIBUTION is used to switch between the old base distribution and distroless base images
+ARG BASE_DISTRIBUTION=default
+
+# -----------------------------------------------------------------------------
+FROM ubuntu:16.04 as default
 RUN apt-get update && apt-get install -y ca-certificates
-RUN chmod 755 /node_agent_k8s
```

Comment:
why remove this line ? it was introduced for ingress-sds in https://github.com/istio/istio/commit/667b31b3c2b38674874ce24c603cf1847650424d#diff-db75d91f6316cdf9e537543775aa557e, @JimmyCYJ does remove this cause regression ?

Reply:
Yes, when I added node agent as ingress gateway agent, I remember there was a permission issue without this line. Please keep it here.

Reply:
Do you know the exact root cause for this issue? Is there a ticket or an integration test related to this problem?
There are many other Dockerfiles, where executables are simply `ADD`ed without running a chmod afterwards. It seems these Dockerfiles are working without this additional chmod.

Reply:
I don't think there is a ticket. Without this line, if SDS is enabled at ingress gateway, the ingress gateway pod will have trouble to run, and kubectl describe pod <ingress gateway pod> shows there is a permission error for node agent. That is all I saw last time. I didn't find other useful information to catch the root cause. If keeping this line does not cause any critical problem, can we keep it? Otherwise, we need to verify removing this line does not break ingressgateway with SDS.

Reply:
> ... If keeping this line does not cause any critical problem, can we keep it? ...

Not a critical problem, but `chmod` is not available on a distroless image per default. This is of course nothing that can't be fixed, but I think having a failing test case first would be essential. Just to see the problem and if it is fixed with chmod alone. Are you planning of such a SDS test?

Reply:
I think this still needs to be fixed, we can't break nodeagent for everyone just for distroless. Agree it should have a test though

Reply:
Distroless or not, chmod should not be necessary.  No test cases fail with a lack of chmod AFAIK.  The reason chmod is unnecessary is docker will install the file with the permissions it was created with.  If the binary is 755, docker installs it as such.

In the interim, could you add chmod to the base image, create an issue to track the need to remove the chmod, and finally indicate the need to investigate why this particular binary needs chmod?  An alternative course of action would be to verify the system works without chmod (via a test case).
Cheers
-steve

Reply:
I agree we should need to use chmod in the docker, we should make sure it is executable before adding it to the image (or if it already is, then do no action).

I just want to make sure we don't break what we have working now, even if the tests pass. I don't know if we have e2e tests for ingressgateway with sds so if we don't it makes sense the tests pass?

Reply:
We tested the SDS case and found no issues with the `distroless` variant

Reply:
Shouldn't be the first action in such cases be to actually write a test to show that/if something is missing? If the passing tests are no indication that it works, what then?

Reply:
Ran into a problem caused by the missing chmod here, see https://github.com/istio/istio/issues/14718.